### PR TITLE
HBASE-27727 Implement filesystem based Replication peer storage

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 20s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 30s |  master passed  |
| +1 :green_heart: |  compile  |   4m 23s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 43s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   5m  9s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 23s |  the patch passed  |
| +1 :green_heart: |  cc  |   4m 23s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 23s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 12s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  12m  7s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  hbaseprotoc  |   2m  6s |  the patch passed  |
| +1 :green_heart: |  spotless  |   0m 39s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   5m 47s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 47s |  The patch does not generate ASF License warnings.  |
|  |   |  54m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux c7f93e4fc714 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 164b8e2600 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 15s |  master passed  |
| +1 :green_heart: |  compile  |   1m 46s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 43s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  8s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 48s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 48s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 59s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 25s |  hbase-protocol-shaded in the patch passed.  |
| -1 :x: |  unit  |   0m 57s |  hbase-client in the patch failed.  |
| +1 :green_heart: |  unit  |   0m 26s |  hbase-replication in the patch passed.  |
| +1 :green_heart: |  unit  | 205m 11s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |  13m  5s |  hbase-mapreduce in the patch passed.  |
|  |   | 246m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux d90107f6a325 5.4.0-1093-aws #102~18.04.2-Ubuntu SMP Wed Dec 7 00:31:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 164b8e2600 |
| Default Java | Temurin-1.8.0_352-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-client.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/testReport/ |
| Max. process+thread count | 2721 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 56s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 34s |  master passed  |
| +1 :green_heart: |  compile  |   2m 18s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 29s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 21s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 17s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 17s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 18s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 39s |  hbase-protocol-shaded in the patch passed.  |
| -1 :x: |  unit  |   1m  8s |  hbase-client in the patch failed.  |
| +1 :green_heart: |  unit  |   0m 35s |  hbase-replication in the patch passed.  |
| +1 :green_heart: |  unit  | 216m 11s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |  14m 15s |  hbase-mapreduce in the patch passed.  |
|  |   | 262m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux aa87d2f4a723 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 164b8e2600 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-client.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/testReport/ |
| Max. process+thread count | 2683 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 20s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  1s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 16s |  master passed  |
| +1 :green_heart: |  compile  |   4m 21s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 40s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   5m  8s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 17s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 21s |  the patch passed  |
| +1 :green_heart: |  cc  |   4m 21s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 21s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  12m  2s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  hbaseprotoc  |   2m  5s |  the patch passed  |
| +1 :green_heart: |  spotless  |   0m 40s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   5m 47s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 46s |  The patch does not generate ASF License warnings.  |
|  |   |  54m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux f3248c9fdf58 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 251e498477 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 81 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 24s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 43s |  master passed  |
| +1 :green_heart: |  compile  |   1m 39s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 33s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 38s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 38s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 33s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 59s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 25s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  8s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 27s |  hbase-replication in the patch passed.  |
| +1 :green_heart: |  unit  | 204m 44s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |  13m 12s |  hbase-mapreduce in the patch passed.  |
|  |   | 245m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux fed3c606206b 5.4.0-1093-aws #102~18.04.2-Ubuntu SMP Wed Dec 7 00:31:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 251e498477 |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/testReport/ |
| Max. process+thread count | 2739 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 53s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 20s |  master passed  |
| +1 :green_heart: |  compile  |   3m 11s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m  7s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 38s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 54s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 54s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 21s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 45s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 38s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 34s |  hbase-replication in the patch passed.  |
| +1 :green_heart: |  unit  | 223m 39s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |  14m 30s |  hbase-mapreduce in the patch passed.  |
|  |   | 280m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 5fc711a60724 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 251e498477 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/testReport/ |
| Max. process+thread count | 2750 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-replication/src/main/java/org/apache/hadoop/hbase/replication/ReplicationPeerStorageType.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+package org.apache.hadoop.hbase.replication;
+
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * Specify the implementations for {@link ReplicationPeerStorage}.
+ */
+@InterfaceAudience.Private
+public enum ReplicationPeerStorageType {
```
Comment (2005hithlj) [inline hbase-replication/src/main/java/org/apache/hadoop/hbase/replication/ReplicationPeerStorageType.java:26]: @Apache9  sir.
Should we keep this enum class as it is not being used?

**Review by 2005hithlj — APPROVED** [review]
LGTM

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 58s |  master passed  |
| +1 :green_heart: |  compile  |   4m 24s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 42s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   5m  8s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 17s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 21s |  the patch passed  |
| +1 :green_heart: |  cc  |   4m 21s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 21s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  12m 11s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  hbaseprotoc  |   2m  6s |  the patch passed  |
| +1 :green_heart: |  spotless  |   0m 40s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   5m 49s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 46s |  The patch does not generate ASF License warnings.  |
|  |   |  55m 10s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux 83482a4758cf 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e4b4cef80e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/3/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 54s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 24s |  master passed  |
| +1 :green_heart: |  compile  |   2m 17s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 26s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 18s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 18s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 38s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 25s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 35s |  hbase-replication in the patch passed.  |
| +1 :green_heart: |  unit  | 216m 49s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |  14m 14s |  hbase-mapreduce in the patch passed.  |
|  |   | 263m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5165 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux ab057657ab22 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e4b4cef80e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/3/testReport/ |
| Max. process+thread count | 2704 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-replication hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5165/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**